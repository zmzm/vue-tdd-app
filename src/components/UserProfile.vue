
<script>
export default {
  name: 'UserProfile'
}
</script>

<template>
  <div>
    UserProfile
  </div>
</template>