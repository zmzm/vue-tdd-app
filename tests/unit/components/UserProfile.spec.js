import { shallowMount } from '@vue/test-utils';

import UserProfile from '@/components/UserProfile';

import userFixture from '../../fixtures/userFixture';

describe('UserProfile', () => {
  let props = {};

  const build = () => {
    const wrapper = shallowMount(UserProfile, {
      props,
    });

    return {
      wrapper,
      avatar: () => wrapper.find('.user-profile__avatar'),
      name: () => wrapper.find('.user-profile__name'),
      bio: () => wrapper.find('.user-profile__bio'),
    };
  };

  beforeEach(() => {
    jest.resetAllMocks();
    props = { user: userFixture };
  });

  it('renders the component', () => {
    // arrange
    const { wrapper } = build();
    // assert
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('renders main components', () => {
    // arrange
    const { avatar, name, bio } = build();
    // assert
    expect(avatar().exists()).toBe(true);
    expect(avatar().attributes().src).toBe(props.user.avatar_url);

    expect(name().exists()).toBe(true);
    expect(name().text()).toBe(props.user.name);

    expect(bio().exists()).toBe(true);
    expect(bio().text()).toBe(props.user.bio);
  });
});
