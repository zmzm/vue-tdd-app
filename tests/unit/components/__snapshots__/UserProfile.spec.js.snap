// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserProfile renders the component 1`] = `
<div class="user-profile"><img class="user-profile__avatar" src="https://github.com/images/error/octocat_happy.gif">
  <div class="user-profile__name">monalisa octocat</div>
  <div class="user-profile__bio">There once was...</div>
</div>
`;
